compiler/rustc_hir_analysis/src/astconv/lint.rs

```diff
@@ -9,8 +9,83 @@ use rustc_trait_selection::traits::error_reporting::suggestions::NextTypeParamNa
 use super::AstConv;
 
 impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
+    /// Prohibit or lint against *bare* trait object types depending on the edition.
+    ///
+    /// *Bare* trait object types are ones that aren't preceeded by the keyword `dyn`.
+    /// In edition 2021 and onward we emit a hard error for them.
+    pub(super) fn prohibit_or_lint_bare_trait_object_ty(
```

Comment:
(esthetics) move the only non-priv item to the top; the other items are all helpers of this one

Reply:
new name is clearer: bare trait? bare trait *object* (ty); furthermore this isn't “linting maybe” but it's first & foremost *prohibiting* (if we assume the latest stable edition to be the canonical edition) or linting — in older editions.